compiler/sem/pragmas.nim

```
@@ -1269,10 +1162,11 @@ proc prepareSinglePragma(
         incl(sym.flags, sfNoalias)
       of wEffectsOf:
         result = processEffectsOf(c, it, sym)
+        if result.isNil: # was there no error?
```

`processEffectsOf` and `processParam` therein are messing this up by silently dropping `semExpr` errors, and they really should create a full production. That doesn't have to be done now; pre-existing issues.